app/src/main/java/org/dhis2/utils/customviews/OptionSetSelectionView.java

```diff
@@ -0,0 +1,236 @@
+                optionBinding.checkBox.setOnCheckedChangeListener((buttonView, isChecked) -> {
+                    if (isChecked) {
+                        listener.onSelectedOption(option.name(), option.code());
+                    } else if (currentCodeValue.equals(option.code())) {
+                        listener.onOptionsClear();
+                    }
+                });
+                checkGroup.addView(optionBinding.getRoot());
+            }
+        }
+        invalidate();
+    }
+
+    private void setRadioOptions() {
+        radioGroup.removeAllViews();
+        for (Option option : options) {
+            if (canShowOption(option.uid())) {
+                OptionSetSelectItemBinding optionBinding = OptionSetSelectItemBinding.inflate(inflater, radioGroup, false);
+                optionBinding.setEditable(isEditable);
+                optionBinding.setOptionName(option.displayName());
+                optionBinding.radio.setChecked(currentCodeValue != null && (currentCodeValue.equals(option.code()) || currentCodeValue.equals(option.name())));
+                optionBinding.radio.setOnCheckedChangeListener((buttonView, isChecked) -> {
+                    if (isChecked) {
+                        listener.onSelectedOption(option.name(), option.code());
+                    }
+                });
+                radioGroup.addView(optionBinding.getRoot());
+            }
+        }
+        invalidate();
+    }
+
+    private boolean canShowOption(String optionUid) {
+        boolean inOptionsToShow = optionsToShow.contains(optionUid);
+        boolean inOptionsToHide = optionsToHide.contains(optionUid);
+        if(!optionsToShow.isEmpty()){
+            return inOptionsToShow;
+        }else{
+            return !inOptionsToHide;
+        }
+    }
+
+    public void setWarning(String warning, String error) {
+        if (!isEmpty(warning)) {
+            warningErrorMessage.setTextColor(ContextCompat.getColor(getContext(), R.color.warning_color));
+            warningErrorMessage.setVisibility(View.VISIBLE);
+        } else if (!isEmpty(error)) {
+            warningErrorMessage.setText(error);
+            warningErrorMessage.setTextColor(ContextCompat.getColor(getContext(), R.color.error_color));
+            warningErrorMessage.setVisibility(View.VISIBLE);
+        } else {
+            warningErrorMessage.setVisibility(View.GONE);
+        }
+    }
+
+    public void setLabel(String label, boolean mandatory) {
+        StringBuilder labelBuilder = new StringBuilder(label);
+        if (mandatory)
+            labelBuilder.append("*");
+        this.label = labelBuilder.toString();
+        labelView.setHint(this.label);
+    }
+
+    public void setDescription(String description) {
+        descriptionLabel.setVisibility(label.length() > 16 || description != null ? View.VISIBLE : View.GONE);
```

Comment:
can you extract the magic number? Thanks ! I think this would not pass klintcheck xD